code/_core/mob/_mob.dm

```diff
@@ -93,6 +93,9 @@
 
 	var/last_z = 0
 
+	var/list/observers = list() //A list of observers/ghosts
```

Comment:
I don't see this variable being .Cut() in Destroy() anywhere, so you should add that.